build-jenkins-agent-ubuntu.pkr.hcl

```diff
@@ -41,6 +41,15 @@ build {
     script           = "./provisioning/ubuntu-provision.sh"
   }
 
+  provisioner "file" {
+    source      = "./goss/goss.yaml"
+    destination = "/tmp/goss.yaml"
+  }
+
+  provisioner "shell" {
+    script           = "/usr/local/bin/goss validate --gossfile /tmp/goss.yaml"
```

Comment:
https://developer.hashicorp.com/packer/docs/provisioners/shell#script you might want to read the documentation ;)